airflow/jobs/scheduler_job.py

```diff
@@ -629,8 +629,9 @@ def _process_task_instances(self, dag, task_instances_list, session=None):
         active_dag_runs = []
         for run in dag_runs:
             self.log.info("Examining DAG run %s", run)
-            # don't consider runs that are executed in the future
-            if run.execution_date > timezone.utcnow():
+            # don't consider runs that are executed in the future unless
+            # specified by config and schedule_interval is None
+            if run.execution_date > timezone.utcnow() and not run.dag.allow_future_exec_dates:
```

Comment:
Just realized this line needs to be changed. Note that run.dag is not yet set at this point. You need to use the `dag` variable. Probably worth considering adding a test for `run.execution_date > timezone.utcnow()` in test_scheduler_job.py to detect this.
```
if run.execution_date > timezone.utcnow() and not dag.allow_future_exec_dates:
```